File: 25-10-2024_Remove_SubFolders_from_the_Filesystem.py
from typing import List

class Solution:
    def removeSubfolders(self, folder: List[str]) -> List[str]:
        # Sort the folder array
        folder.sort()
        
        # List to store root folders
        rootsFolder = [folder[0]]  # Add the first folder as a root

        # Iterate through the sorted folder list
        for i in range(1, len(folder)):
            lastRoot = rootsFolder[-1]  # Get the last added root folder
            # Check if the current folder is not a subfolder of the last added root
            if not folder[i].startswith(lastRoot) or folder[i][len(lastRoot)] != '/':
                rootsFolder.append(folder[i])  # Add to rootsFolder

        return rootsFolder  # Return the resulting list of strings

File: 25-10-2024_Remove_SubFolders_from_the_Filesystem.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

class Solution {
    public List<String> removeSubfolders(String[] folder) {
        // Sort the folder array
        Arrays.sort(folder);
        
        // List to store root folders
        List<String> rootsFolder = new ArrayList<>();
        rootsFolder.add(folder[0]); // Add the first folder as a root

        // Iterate through the sorted folder list
        for (int i = 1; i < folder.length; i++) {
            String lastRoot = rootsFolder.get(rootsFolder.size() - 1);
            // Check if the current folder is not a subfolder of the last added root
            if (!folder[i].startsWith(lastRoot) || 
                folder[i].charAt(lastRoot.length()) != '/') {
                rootsFolder.add(folder[i]); // Add to rootsFolder
            }
        }

        return rootsFolder; // Return the resulting list of strings
    }
}
